warehouse/static/js/warehouse/controllers/gitlab_repo_stats_controller.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static outlets = ["gitlab-repo-info"];
+  static values = {
+    repository: String,
+  };
+
+  connect() {
+    this.load();
+  }
+
+  load() {
+    const get_project_objects_count = (project_api_url, project_object, fetchParams) => {
+      return fetch(
+        `${project_api_url}/${project_object}?` + new URLSearchParams({
+          "state": "opened",
+          "per_page": 1,
+        }),
+        fetchParams
+      ).then((response) => {
+        if(response.ok) {
+          return response.json().then(
+            (data) => Object.keys(data).length ? response.headers.get("X-Total-Pages") : 0
+          );
+        }
+
+        return 0;
+      });
+    };
+
+    const GITLAB_API_URL = "https://gitlab.com/api/v4";
+    const project_id = encodeURIComponent(this.repositoryValue);
+    const fetchParams = {
+      method: "GET",
+      mode: "cors",
+    };
+
+    const project_api_url = `${GITLAB_API_URL}/projects/${project_id}`;
+    const project_info = fetch(
+      project_api_url,
+      fetchParams
+    ).then((response) =>
+      response.ok === true ? response.json() : null
+    );
+
+    const issues = get_project_objects_count(project_api_url, "issues", fetchParams);
+    const merge_requests = get_project_objects_count(project_api_url, "merge_requests", fetchParams);
+
+    Promise.all([project_info, issues, merge_requests]).then(
+      ([project, issues_count, merge_requests_count]) => {
+        const stats = {
+          issues: issues_count,
+          issues_url: project.web_url + "/-/issues",
+          followers: project.star_count || 0,
+          followers_url: project.web_url + "/-/starrers",
+          forks: project.forks_count || 0,
+          forks_url: project.web_url + "/-/forks",
+          MRs: merge_requests_count,
+          MRs_url: project.web_url + "/-/merge_requests",
+        };
+        this.gitlabRepoInfoOutlets.forEach((outlet) =>
+          outlet.updateStats(stats)
+        );
+      })
+      // swallow errors, we don't want to show them to the user
+      .catch(() => {});
```

Comment:
We don't want to show them to the user, but would probably be helpful to at least log them to the console here.